myFslInstallerScript.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+#myfslinstaller.sh
+
+set -e
+set -u
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# input args (with defaults):
+# * version
+# * os (centos 5 to 7)
+# * install dir
+# * download dir
+
+#myfslinstaller function takes the parameters (OS and Version) and downloads the corresponding file of fsl installer from the server.
+if [[ $# -eq 0 ]] ; then
+    echo 'No parameters are passed, taking default values'
+fi
+
+version=${1:-5.0.6}
+os=${2:-CentOS7}
+#user=$USER
+installDir=${3:-/usr/local/src}
+downloadDir=${4:-/usr/local/etc/}
+
+function myFslInstaller {
+echo "###----FSL INSTALLER---###"
+echo "Version of FSL:$version"
+echo "OS:$os"
+#echo "Username:$user"
+echo "Installation Directory:$installDir"
+echo "Download Directory: $downloadDir"
+
+if [[ ! ("$version" =~ ^[0-9].+$) ]]; then 
+    echo 'Please pass a number that corresponds to an existing version of FSL'
```

Comment:
Your test doesn't check that. I would write `This doesn't look like a valid version number.` instead.